x-pack/plugins/security_solution/public/resolver/view/process_event_dot.tsx

```diff
@@ -293,25 +295,14 @@ const ProcessEventDotComponents = React.memo(
     // Entity ID of self
     const selfEntityId = eventModel.entityId(event);
 
-    const isShowingEventActions = magFactorX > 0.8;
-    const isShowingDescriptionText = magFactorX >= 0.55;
+    const isShowingEventActions = xScale > 0.8;
+    const isShowingDescriptionText = xScale >= 0.55;
 
     /**
      * As the resolver zooms and buttons and text change visibility, we look to keep the overall container properly vertically aligned
      */
-    const actionalButtonsBaseTopOffset = 5;
-    let actionableButtonsTopOffset;
-    switch (true) {
-      case isShowingEventActions:
-        actionableButtonsTopOffset = actionalButtonsBaseTopOffset + 3.5 * magFactorX;
-        break;
-      case isShowingDescriptionText:
-        actionableButtonsTopOffset = actionalButtonsBaseTopOffset + magFactorX;
-        break;
-      default:
-        actionableButtonsTopOffset = actionalButtonsBaseTopOffset + 21 * magFactorX;
-        break;
-    }
+    const actionableButtonsTopOffset =
+      (isShowingEventActions ? 3.5 : isShowingDescriptionText ? 1 : 21) * xScale + 5;
```

Comment:
Nice, you don't have anything against nested ternary's from a readability standpoint?

Reply:
wanted your thoughts on this (esp. since you wrote the original.)

1. I found the original a lot to read due to boilerplate. Specifically, reading `actionableButtonsTopOffset` vs `actionableButtonsTopOffset` was hard.
2. The repeated part of `actionalButtonsBaseTopOffset + ` and `* magFactorX` required me to visually compared the 3 versions to see that they were the same.
3. part of my motivation to make this terse is because the function it is in is long, and has many variables that aren't related to each other. That isn't the best motivation to make something terse. We could move the logic out of the function instead.

Reply:
1. agreed, makes sense
2. also makes sense
3/4. I like to throw in the occasional switch every now and again 😂 
5. Even if it's terse, it's still readable, so no need to revert any of it. I like moving it into an external function though because it will be a better separation of concerns and eventually it can probably be removed altogether in favor of mostly pure css.